File: scaling/cpp/sampleconfiguration.cpp
// Copyright (c) Microsoft. All rights reserved.

#include "pch.h"
#include "MainPage.xaml.h"
#include "SampleConfiguration.h"

using namespace SDKTemplate;

Platform::Array<Scenario>^ MainPage::scenariosInner = ref new Platform::Array<Scenario>
{
    { "Load images for different scales",         "Scaling.Scenario1" },
    { "Override default scaling of UI elements",  "Scaling.Scenario2" },
};

File: scaling/cs/sampleconfiguration.cs
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

using System;
using System.Collections.Generic;
using Windows.UI.Xaml.Controls;
using Scaling;

namespace SDKTemplate
{
    public partial class MainPage : Page
    {
        public const string FEATURE_NAME = "Scaling";

        List<Scenario> scenarios = new List<Scenario>
        {
            new Scenario() { Title = "Load images for different scales",         ClassType = typeof(Scaling.Scenario1) },
            new Scenario() { Title = "Override default scaling of UI elements",  ClassType = typeof(Scaling.Scenario2) },
        };
    }

    public class Scenario
    {
        public string Title { get; set; }
        public Type ClassType { get; set; }
    }
}
